api/src/main/java/ca/bc/gov/educ/api/batchgraduation/listener/UserReqDistributionRunCompletionNotificationListener.java

```diff
@@ -137,17 +136,23 @@ private void processGlobalList(DistributionSummaryDTO summaryDTO, Long batchId,
 				activityCode = "OT".equalsIgnoreCase(credentialType) ? "USERDISTOT" : "USERDISTRC";
 			}
 			if(!studentCredentialDistributionControlList.isEmpty()) {
+				DistributionResponse disres = null;
 				DistributionRequest distributionRequest = DistributionRequest.builder().mapDist(mapDist).activityCode(activityCode).studentSearchRequest(summaryDTO.getStudentSearchRequest()).build();
 				if (credentialType.equalsIgnoreCase("RC")) {
 					disres = restUtils.createReprintAndUpload(batchId, accessToken, distributionRequest, activityCode, localDownload);
 				} else {
 					disres = restUtils.mergeAndUpload(batchId, distributionRequest, activityCode, localDownload);
 				}
 				if(disres != null) {
-					updateBackStudentRecords(studentCredentialDistributionControlList.stream().distinct().toList(),batchId,activityCode);
+					LOGGER.info("Merge and Upload Status {}",disres.getMergeProcessResponse());
+					 if(!FAILED.name().equals(disres.getMergeProcessResponse())) {
+						 updateBackStudentRecords(studentCredentialDistributionControlList.stream().distinct().toList(),batchId,activityCode);
+						 return true;
+					 }
 				}
 			}
 		}
+		return false;
```

Comment:
Always a failed run if credentialType is null? I think an error log would be helpful for this scenario